api_app/analyzers_manager/file_analyzers/malcore_scan.py

```diff
@@ -0,0 +1,333 @@
+# This file is a part of IntelOwl https://github.com/intelowlproject/IntelOwl
+# See the file 'LICENSE' for copying permission.
+
+import logging
+import time
+
+import requests
+
+from api_app.analyzers_manager.classes import FileAnalyzer
+from api_app.analyzers_manager.exceptions import AnalyzerRunException
+from tests.mock_utils import MockUpResponse, if_mock_connections, patch
+
+logger = logging.getLogger(__name__)
+
+
+class MalcoreScan(FileAnalyzer):
+    base_url = "https://api.malcore.io/api/"
+    _api_key_name: str
+    poll_distance = 4
+
+    def run(self):
+        self.headers = {"apiKey": self._api_key_name}
+
+        binary = self.read_file_bytes()
+        files = {"filename1": binary}
+        try:
+            logger.info(f"Sending {self.md5} to Malcore")
+            response = requests.post(
+                self.base_url + "upload", headers=self.headers, files=files
+            )
+            response.raise_for_status()
+        except requests.RequestException as e:
+            raise AnalyzerRunException(e)
+
+        try:
+            uuid = response.json()["data"]["data"]["uuid"]
+        except TypeError:
+            raise AnalyzerRunException(f"Failed analysis for {self.md5}")
+
+        for _try in range(self.max_tries):
+            logger.info(f"polling malcore try #{_try + 1}")
+            self.result = self._get_status(uuid)["data"]
+            if "status" not in self.result and "msg" not in self.result:
+                break
+
+            time.sleep(self.poll_distance)
```

Comment:
max_tries and poll_distance must be defined both as a class attribute and in the migration file as params.